autosubmit/job/job_list.py

```diff
@@ -3045,16 +3045,16 @@ def remove_rerun_only_jobs(self, notransitive=False):
             self.update_genealogy()
         del self._dic_jobs
 
-    def print_with_status(self, status_change=None, nocolor=False, existing_list=None) -> str:
+    def print_with_status(self, status_change: Optional[dict[Any, Any]] = None, nocolor=False, existing_list=None) -> str:
```

Comment:
Fix for #2330 , and the rest are IDE warnings (as the test is covering this, good opportunity to clean up some more warnings)